src/creatures/creature.cpp

```diff
@@ -448,7 +464,7 @@ void Creature::checkSummonMove(const Position& newPos, bool teleportSummon) cons
 {
 	if (hasSummons()) {
 		// Check if any of our summons is out of range (+/- 2 floors or 30 tiles away)
```

Comment:
I think there's something missing on this check, because after you summon 2 creatures, and go out of range, they never disappear and you're not able to summon any other creatures unless they got killed. (Main issue too)